### ISSUE TITLE
Worktrees are not recognized in WSL

### ISSUE THREAD

**Issue by tomasorti** [sublimehq/sublime_merge#1242]
**Version info**

 - OS: [Windows 10]
 - Build: [2059]

**Description**
I am working in WSL (Windows Subsystem for Linux).
If I try to load a worktree path via `File | Open Repository` and select the develop path, I get a Windows Dialog with the text:

`~/my-project/develop doesn’t look like a git repository.
`
This might be the reason it pops the Windows Dialog…
The .git in the worktree directory in WSL it is not a directory, neither a symlink, but a text file…

More details with an real repo example can be found in this forum post:
https://forum.sublimetext.com/t/how-does-worktree-work-in-sublime-merge/59011

**Steps to reproduce**
1. Try it with a repository under WSL
2. Use `File | Open Repository` and select the develop path,
3. You will get `~/my-project/develop doesn’t look like a git repository.`

**Expected behavior**
SM should handle the git worktree

**Comment by stinos** [comment]
What do you mean with 'working *in* WSL' ? It's a bit confusing because later on you mention a Windows Dialog being shown but you do mention unix-style paths a well. So I'm wondering: do you run the Windows version of SM and open a path which happens to be in `\\wsl$` (can be tricky wrt symlinks etc), or do you run the unix version of SM in WSL?

**Comment by tomasorti** [comment]
When I said _working in WSL_, I meant I use git in the WSL command line.
I created the worktree with `git worktree add` under WSL terminal.
And I use Windows version of SM...

This is the workflow I normally use.
WSL terminal gives me Linux capabilities which I need for my project that do not give CMD or Powershell.
And I use SM on Windows as the git UI for better diff, etc...

I can see where you may be pointing... to use unix version SM on WSL... Right?
Tend to use as many tools as I can in native form.

**Comment by stinos** [comment]
Ok thanks for explaining. Can you also clarify which path exactly you open with `File | Open repository`? How do you access the WSL filesystem?

> I can see where you may be pointing... to use unix version SM on WSL... Right?

Yes that's what I do because there can be issues because of filesystem differences when mixing things, plus it's much faster (could depend on WSL version though). I use XLaunch for the Windows site, it's close to flawless.

**Comment by tomasorti** [comment]
> Can you also clarify which path exactly you open with File | Open repository? How do you access the WSL filesystem?
The path is the Windows normal path, I do not access WSL filesystem.

The reason I started using SM is because git for Windows really worked badly.
On Linux I would use Emacs with magit, which was my preferred way, but Emacs on Windows using git for Windows sucks.
It is painfully slow.

And I also tried Emacs on WSL (with XLaunch) with WSL git, but also kind of slow.
Even though X stuff works, they tend to be slower, and I prefer to use a native UI in the OS I am using.
Anyway, I will give a try to SM on WSL.
